src/registrar/models/domain.py

```diff
@@ -668,6 +654,210 @@ def __str__(self) -> str:
+        """Abstracts the cache logic on EppLib contact items
+
+        contact_type_choice is a literal in PublicContact.ContactTypeChoices,
+        for instance: PublicContact.ContactTypeChoices.SECURITY.
+
+        If you wanted to setup getter logic for Security, you would call:
+        cache_contact_helper(PublicContact.ContactTypeChoices.SECURITY),
+        or cache_contact_helper("security")
+        """
+        try:
+            # TODO - refactor
+            desired_property = "contacts"
+            # The contact type 'registrant' is stored under a different property
+            if contact_type_choice == PublicContact.ContactTypeChoices.REGISTRANT:
+                desired_property = "registrant"
+            contacts = self._get_property(desired_property)
+            if contact_type_choice == PublicContact.ContactTypeChoices.REGISTRANT:
+                contacts = [contacts]
+        except KeyError as error:
+            logger.warning("generic_contact_getter -> Contact does not exist")
+            logger.warning(error)
+            # Should we just raise an error instead?
+            return self.get_contact_default(contact_type_choice)
+        else:
+            print(f"generic_contact_getter -> contacts?? {contacts}")
+            # --> Map to public contact
+            cached_contact = self.grab_contact_in_keys(contacts, contact_type_choice)
+            if cached_contact is None:
+                raise ValueError("No contact was found in cache or the registry")
+
+            # Convert it from an EppLib object to PublicContact
+            return cached_contact
+
+    def get_default_security_contact(self):
+        """Gets the default security contact."""
+        contact = PublicContact.get_default_security()
+        contact.domain = self
+        return contact
+
+    def get_default_administrative_contact(self):
+        """Gets the default administrative contact."""
+        contact = PublicContact.get_default_administrative()
+        contact.domain = self
+        return contact
+
+    def get_default_technical_contact(self):
+        """Gets the default administrative contact."""
+        contact = PublicContact.get_default_technical()
+        contact.domain = self
+        return contact
+
+    def get_default_registrant_contact(self):
+        """Gets the default administrative contact."""
+        contact = PublicContact.get_default_registrant()
+        contact.domain = self
+        return contact
+
+    def grab_contact_in_keys(self, contacts, check_type):
+        """Grabs a contact object.
+        Returns None if nothing is found.
+        check_type compares contact["type"] == check_type.
+
+        For example, check_type = 'security'
+        """
+        for contact in contacts:
+            print(f"grab_contact_in_keys -> contact item {contact.__dict__}")
+            if (
+                isinstance(contact, PublicContact)
+                and contact.registry_id is not None
```

Comment:
why this? if registry_id is none should an error be thrown instead?

Reply:
That is a good point. Throwing an error would be better here